common/rfb/VNCSConnectionST.cxx

```diff
@@ -401,6 +401,14 @@ bool VNCSConnectionST::needRenderedCursor()
   return false;
 }
 
+void VNCSConnectionST::clientInitReady()
+{
+  if (state() != RFBSTATE_INITIALISATION)
+    return;
+
+  clientInit(shared_);
```

Comment:
I think this design is a bit confusing. `clientInit()` is a callback for a specific message in the protocol. It should only be called when we get that message.

An intermediate step is probably needed after `clientInit()` has been called where we wait for the desktop.